workflow-service/src/test/java/com/redhat/parodos/workflow/definition/service/WorkFlowDefinitionServiceImplTest.java

```diff
@@ -109,6 +131,42 @@ public void simpleSaveTest() {
 		assertEquals(argument.getValue().getWorkFlowTaskDefinitions().stream().findFirst().get().getName(), TEST_TASK);
 	}
 
+	@Test
+	public void simpleSaveTest_skipSave_when_sameWorkflowFoundInDB() {
+		String workFlowName = "test";
+		String workFlowTaskName = "testTask";
+		// given
+		WorkFlowDefinition workFlowDefinition = this.sampleWorkFlowDefinition(workFlowName);
+		workFlowDefinition.setParameters("{}");
+		WorkFlowTask workFlowTask = Mockito.mock(WorkFlowTask.class);
+		WorkFlowTaskParameter workFlowTaskParameter = WorkFlowTaskParameter.builder().key("key").description("the key")
+				.optional(false).type(WorkFlowTaskParameterType.URL).build();
+		Mockito.when(workFlowTask.getName()).thenReturn(workFlowTaskName);
+		Mockito.when(workFlowTask.getWorkFlowTaskParameters()).thenReturn(List.of(workFlowTaskParameter));
+		WorkFlowTaskDefinition workFlowTaskDefinition = this.sampleWorkFlowTaskDefinition(workFlowDefinition,
+				workFlowTaskName, workFlowTaskParameter);
+		workFlowTaskDefinition.setParameters("{}");
+		workFlowTaskDefinition.setOutputs("[]");
+		workFlowDefinition.setWorkFlowTaskDefinitions(List.of(workFlowTaskDefinition));
+		Mockito.when(this.workFlowTaskDefinitionRepository.save(any())).thenReturn(workFlowTaskDefinition);
+		Mockito.when(this.workFlowDefinitionRepository.save(any())).thenReturn(workFlowDefinition);
+		Mockito.when(this.workFlowDefinitionRepository.findFirstByName(anyString())).thenReturn(workFlowDefinition);
+		Mockito.when(this.workFlowTaskDefinitionRepository.findFirstByName(anyString()))
+				.thenReturn(workFlowTaskDefinition);
+		// when
+		WorkFlowDefinitionResponseDTO workFlowDefinitionResponseDTO = this.workFlowDefinitionService.save(workFlowName,
+				WorkFlowType.ASSESSMENT, Collections.emptyList(), List.of(workFlowTask),
+				WorkFlowProcessingType.SEQUENTIAL);
+
+		// then
+		assertNotNull(workFlowDefinitionResponseDTO);
+		assertNotNull(workFlowDefinitionResponseDTO.getId());
+		assertEquals(workFlowDefinitionResponseDTO.getName(), workFlowName);
+
+		Mockito.verify(this.workFlowDefinitionRepository, Mockito.times(1)).save(any());
```

Comment:
Do you think It's possible to assert the parameter of `save()` as well? I mean to use `eq(..)` instead of `any()` ? In this way, we can test that there aren't unwanted edit to the object.